contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -719,12 +752,21 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
             _wethSharePct < allowedWethShareStart ||
             _wethSharePct > allowedWethShareEnd
         ) {
+            if (dryRun) {
+                return false;
+            }
+
             revert PoolRebalanceOutOfBounds(
                 _wethSharePct,
                 allowedWethShareStart,
                 allowedWethShareEnd
             );
         }
+
+        // return before emitting the event is intentional
+        if (dryRun) {
+            return true;
```

Comment:
I'd get rid of the `if` statement, and let it always return `true` when it's within the bounds

Reply:
Then it would behave the same with the `dryRun` being true or false. I didn't want the `PoolRebalanced` event to be emitted when `dryRun` equals `true`

Reply:
Agree on not emitting the event.

Reply:
Moved the event emission into the `_rebalance` function. The rebalance event now more appropriately only gets emitted once now: https://github.com/OriginProtocol/origin-dollar/pull/2266/commits/c7346e63a48e9658dc1973327ff93e3b7f959998